kuma/javascript/src/bcd-signal.jsx

```diff
@@ -0,0 +1,829 @@
+
+        stepsInfo.appendChild(stepsInfoSpan);
+        formControlBlockWrapper.appendChild(stepsInfo);
+
+        formControlBlockWrapper.appendChild(signalStepOneBlock());
+        formControlBlockWrapper.appendChild(signalStepTwoBlock());
+
+        return formControlBlockWrapper;
+    };
+
+    /**
+     * Builds and returns form first step with browser select and row select controls
+     * @returns Form first step as a `HTMLElement`
+     */
+    const signalStepOneBlock = () => {
+        const signalStepOneBlock = document.createElement('div');
+        const controls = document.createElement('div');
+        const stepsButtonBlock = document.createElement('div');
+        const nextStepButtonIcon = document.createElement('span');
+        nextStepButton = document.createElement('button');
+
+        signalStepOneBlock.className = 'inner-step';
+        signalStepOneBlock.id = 'step-1';
+        controls.className = 'controls';
+        stepsButtonBlock.className =
+            'navigation-buttons reverse mob-reduced-space';
+        nextStepButton.className = 'button neutral disabled next-step-btn';
+
+        nextStepButton.addEventListener('click', () => {
+            toStep(2);
+        });
+        nextStepButton.innerText = gettext('Next step (2 of 2)');
+
+        nextStepButtonIcon.className = 'icon-next';
+        nextStepButton.appendChild(nextStepButtonIcon);
+
+        stepsButtonBlock.appendChild(nextStepButton);
+        controls.appendChild(createSelectBrowserControl());
+        controls.appendChild(createSelectRowControl());
+        signalStepOneBlock.appendChild(controls);
+        signalStepOneBlock.appendChild(stepsButtonBlock);
+
+        return signalStepOneBlock;
+    };
+
+    /**
+     * Builds and returns form second step with brieft description and supporting material controls
+     * @returns Form second step as a `HTMLElement`
+     */
+    const signalStepTwoBlock = () => {
+        const signalStepTwoBlock = document.createElement('div');
+        const controls = document.createElement('div');
+        const stepsButtonBlock = document.createElement('div');
+        const goBackButton = document.createElement('button');
+        const goBackIcon = document.createElement('span');
+        sendReportButton = document.createElement('button');
+
+        signalStepTwoBlock.id = 'step-2';
+        sendReportButton.className =
+            'button neutral disabled main-btn scroll-to-signal';
+        sendReportButton.innerText = gettext('Send report');
+        signalStepTwoBlock.className = 'inner-step';
+        controls.className = 'controls';
+        stepsButtonBlock.className = 'navigation-buttons reverse';
+        goBackButton.className = 'button prev-step-btn btn-dark';
+        goBackIcon.className = 'icon-back';
+        const backBtnLabel = document.createTextNode(gettext('Previous step'));
+
+        goBackButton.appendChild(goBackIcon);
+        goBackButton.appendChild(backBtnLabel);
```

Comment:
It's due to the nature of the content, that needs to be appended to the button. In this case we had an icon and then a text to be appended to it. I couldn't alter the content of the button directly with innerText, since it has two elements inside. The createTextNode function allows to specify an initial text that is added to that node. It avoids multiple steps that createElement('whatever') needs and outputs a simple text node without a <span> or other tags to accompany it. The last step is appending the icon and the text node to the button. 

Do you prefer the other option, for consistency?

Reply:
Nah. Leave it as is.